Userland/Libraries/LibBencode/Bencode.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2021, Cesar Torres <shortanemoia@protonmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/OwnPtr.h>
+#include <AK/StdLibExtras.h>
+#include <AK/String.h>
+#include <AK/StringBuilder.h>
+#include <AK/Tuple.h>
+#include <AK/Variant.h>
+#include <AK/Vector.h>
+
+namespace Bencode {
+class BencodeEncoder {
+public:
+    void append(i64 value);
+    void append(StringView const& value);
+    void append(ReadonlyBytes const& value);
+    void start_list();
+    void start_dict();
+    void end_list();
+    void end_dict();
+    NonnullOwnPtr<ByteBuffer> to_buffer();
+    void clear();
+
+private:
+    StringBuilder m_builder {};
+};
+
+enum class BencodeNodeType {
+    Integer,
+    String,
+    List,
+    Dictionary,
+    KeyValuePair
+};
+
+struct BencodeNode;
+
+struct KeyValuePair {
+    String key;
+    OwnPtr<BencodeNode> value;
+};
+
+struct BencodeNode {
+    BencodeNodeType type;
+    Variant<i64, String, OwnPtr<KeyValuePair>, OwnPtr<Vector<OwnPtr<BencodeNode>>>> data;
```

Comment:
If the `OwnPtr<BencodeNode>` will never be null (which I assume, as it's very rare to want to store null pointers in a `Vector`, you could use `OwnPtr<NonnullOwnPtrVector<BencodeNode>>` here. Without knowing what you are trying to do, it probably also doesn't make sense to be able to store a null `OwnPtr` in the variant, so maybe you could change those to `NonnullOwnPtr` too.